sdk/communication/Azure.Communication.Rooms/src/autorest.md

```diff
@@ -0,0 +1,22 @@
+# Azure.Communication.CallingServer
+
+When a new version of the swagger needs to be updated:
+1. Open Azure.Communication.sln and under sdk\communication, run `dotnet msbuild /t:GenerateCode` to generate code.
+
+2. Upload the Azure.Communication.Call.dll to the [apiview.dev tool](https://apiview.dev/).
+If any of the new objects needs to be overwritten, add the required changes to the 'Models' folder.
+
+3. Repeat 2 and 3 until the desided interface is reflected in the apiview.dev.
+
+### AutoRest Configuration
+> see https://aka.ms/autorest
+
+```yaml
+title: Rooms
+model-namespace: false
+require:
+    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/e30976f6ccb058a36cd2f9d5160e1fd51f6c5d95/specification/communication/data-plane/Rooms/readme.md
+payload-flattening-threshold: 10
```

Comment:
What's this for?  I don't think we even support flattening in modern AutoRest backends.

Reply:
I tired removing this parameter and regenerating the files to see what it does.
The client is updated to accept request object instead of accepting parameters. Also, classes which were earlier internal changed to public.
 
-        internal HttpMessage CreateUpdateRoomRequest(string roomId, DateTimeOffset? validFrom, DateTimeOffset? validUntil, RoomJoinPolicy? roomJoinPolicy, IEnumerable<RoomParticipantInternal> participants)
- 
+        internal HttpMessage CreateUpdateRoomRequest(string roomId, UpdateRoomRequest patchRoomRequest)

I am not sure if this is preferred for Dotnet SDK, as I see other projects have this parameter and client methods accept individual parameters.
Please do let me know what the general recommendations are. 

Not changing this for public preview but will consider direction for upcoming releases.

Reply:
I haven't seen many folks doing this, but I think it's fine if that's the API shape you want.